etna/records/tests/test_views.py

```diff
@@ -722,3 +723,72 @@ def test_invalid_response_from_kong_raises_404(self):
         response = self.client.get("/records/images/C123456/11000000000000000000/")
 
         self.assertEquals(response.status_code, 404)
+
+
+class RecordDetailBackToSearchTest(TestCase):
+    @responses.activate
+    def test_back_to_search_context_with_catalogue_search(self):
+        expected_url = "/search/catalogue/?sort_by=title&q=london&filter_keyword=paper&level=Item&collection=ADM&collection=BT&closure=Open+Document%2C+Open+Description&opening_start_date_0=&opening_start_date_1=&opening_start_date_2=1900&opening_end_date_0=&opening_end_date_1=&opening_end_date_2=2020&per_page=20&sort_order=asc&display=list&page=2&group=tna"
+
+        responses.add(
+            responses.GET,
+            "https://kong.test/data/fetch",
+            json=create_response(
+                records=[
+                    create_record(iaid="C13359805"),
+                ]
+            ),
+        )
+        record_detail_url = reverse(
+            "details-page-machine-readable", kwargs={"iaid": "C13359805"}
+        )
+        session = self.client.session
+        session["back_to_search_url"] = expected_url
+        session.save()
+        response = self.client.get(record_detail_url)
+        expected_label = "Back to search results"
+        self.assertEqual(response.context.get("back_to_search_label"), expected_label)
+        self.assertEqual(response.context.get("back_to_search_url"), expected_url)
```

Comment:
**Suggestion**: You could swap these two assertions for a an [assertContains](https://docs.djangoproject.com/en/4.2/topics/testing/tools/#django.test.SimpleTestCase.assertContains) to gain additional assurance that the template is using the correct variables and rendering the button successfully.

Reply:
- Have updated the tests for assertContains, however  `parsed_url = parse.quote(browser_search_url, safe="")` does not feel right for a scenario with query params - `def test_back_to_search_render_with_catalogue_search` .This is regardless of using urllib.parse.quote/unquote to set or decode the saved search.
- Unsure about another way to run this test without encoding the saved url in the unit test.

Reply:
Why not just recreate in local dev and copy the generated HTML into a variable?